lib/data-types.js

```diff
@@ -39,24 +42,47 @@ var util = require('util')
  */
 
 var ABSTRACT = function(options) {
-
 };
 
 ABSTRACT.prototype.dialectTypes = '';
 
-ABSTRACT.prototype.toString = function() {
-  return this.toSql();
+ABSTRACT.prototype.toString = function(options) {
+  return this.toSql(options);
 };
 ABSTRACT.prototype.toSql = function() {
   return this.key;
 };
-ABSTRACT.prototype.warn = function(text) {
-  if (!warnings[text]) {
-    warnings[text] = true;
-    console.warn('>> WARNING:', text, '\n>> Check:', this.dialectTypes);
+ABSTRACT.warn = function(link, text) {
+  deprecate(text + '\n>> Check: ' + link);
```

Comment:
Agree, should be a warning more than a deprecation since we do want to normalize, but people should be aware that stuff might happen.